File: 0007_Reverse Integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        ans = 0
        symbol = 1
        int_max = 2 ** 31 - 1
        int_min = -(2 ** 31)
        if x < 0:
            x = -x
            symbol = -1
        while x != 0:
            if ans > 0 and ans > (int_max - x % 10) // 10:
                return 0
            if ans < 0 and ans < (int_min - x % 10) // 10:
                return 0
            ans = ans * 10 + x % 10
            x = x // 10
            
        return ans * symbol

File: 0007_Reverse Integer/solution.go
func reverse(x int) int {
    var ans int
    for x != 0 {
        if ans > 0 && ans > (math.MaxInt32 - x % 10) / 10 {
            return 0
        }
        if ans < 0 && ans < (math.MinInt32 - x % 10) / 10 {
            return 0
        }
        ans = ans * 10 + x % 10
        x = x / 10
    }
    return ans
}